src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1221,6 +1221,44 @@ void MPS::full_state_vector_internal(cvector_t& statevector,
   statevector = reverse_all_bits(temp_statevector, num_qubits);
 }
 
+void MPS::get_amplitude_vector(const reg_t &base_values, cvector_t &amplitude_vector) {
+  move_all_qubits_to_sorted_ordering();
+  uint_t num_values = base_values.size();
+  std::string base_value;
+#pragma omp parallel for if (num_values > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
+  for (int_t i=0; i<static_cast<int_t>(num_values); i++) {
+    base_value = AER::Utils::int2string(base_values[i]);
+    amplitude_vector[i] = get_single_amplitude(base_value);
+  }
+}
+
+complex_t MPS::get_single_amplitude(const std::string &base_value) {
+  // We take the bits of the base value from right to left in order not to expand the 
+  // base values to the full width of 2^n
+  // We contract from left to right because the representation in Qiskit is from left 
+  // to right, i.e., 1=1000, 2=0100, ...
+
+  int_t pos = base_value.length()-1;
+  uint_t bit = base_value[pos]=='0' ? 0 : 1;
+  pos--;
+  cmatrix_t temp = q_reg_[0].get_data(bit);
+
+  for (int_t qubit=0; qubit<num_qubits_-1; qubit++) {
+    if (pos >=0)
+      bit = base_value[pos]=='0' ? 0 : 1;
+    else
+      bit = 0;
+    for (uint_t row=0; row<temp.GetRows(); row++)
+      for (uint_t col=0; col<temp.GetColumns(); col++){
+	temp(row, col) *= lambda_reg_[qubit][col];
```

Comment:
identation

Reply:
Added '{', '}', because in my editor the indentation looks okay. Maybe the braces will help.